src/main/scala/abalone/variant/Variant.scala

```diff
@@ -102,6 +103,144 @@ abstract class Variant private[variant] (
         )
     }.toMap
 
+  def validMovesOf2(situation: Situation, validMoves1: Map[Pos, List[Move]]): Map[Pos, List[(String, Move)]] =
+    situation.board.piecesOf(situation.player).flatMap {
+      case ((pos, piece)) => {
+        Map(
+          pos -> pos.neighbours.flatMap {
+            case Some(neighbour) if(situation.board.piecesOf(situation.player).contains(neighbour)) => {
+              neighbour.neighbours.flatMap {
+                case Some(neighbourOfNeighbour)
+                  if (situation.board.isEmptySquare(Some(neighbourOfNeighbour))
+                  && (pos.isInLine(neighbour, neighbourOfNeighbour) || situation.board.isEmptySquare(pos.dir(neighbour.dir(neighbourOfNeighbour))))
```

Comment:
I'm not sure i understand why we do the second half of this line (`situation.board.isEmptySquare(pos.dir(neighbour.dir(neighbourOfNeighbour)))`)

Reply:
it is to handle that case : 
```
   *  *  Z  *  *
 *  *  a  b  *  *
*  *  *  *  *  *  *  
```
dir can be applied to b but not applied to a

Reply:
(if we move upRight)

Reply:
with `if (situation.board.isEmptySquare(Some(neighbourOfNeighbour))`
neighbourOfNeighbour could take the value of Z